source/base/mpi.cc

```diff
@@ -1033,6 +1033,32 @@ namespace Utilities
     }
 
 
+    namespace internal
+    {
+      template <typename T>
+      std::enable_if_t<is_mpi_type<T>::value, T>
+      builtin_single_broadcast(const MPI_Comm &   comm,
+                               T                  object_to_send,
+                               const unsigned int root_process)
```

Comment:
You will still need explicit template instantiations. Without that, I would expect undefined references in new test cases.

Reply:
Could you explain why we need explicit template instantiation?

Reply:
The definition of this function is only in the `*.cc` file and its visibility is limited to this file (since it's not a header file).
Consequently, you get implicit template instantiations for all template parameters that use this function within this file but you will get undefined references for all other types since the header file doesn't contain the definition but only a declaration of this function.

Reply:
Thank you for the explanation, it is clearer now. I wanted to know by myself when the explicit instantiation is needed